src/shared/tests.h

```diff
@@ -201,41 +201,46 @@ static inline int run_test_table(void) {
 #define DEFINE_TEST_MAIN(log_level)                     \
         DEFINE_TEST_MAIN_FULL(log_level, NULL, NULL)
 
-#define ASSERT_OK(expr)                                                                                  \
-        ({                                                                                               \
-                int _result = (expr);                                                                    \
-                if (_result < 0) {                                                                       \
-                        log_error_errno("Assertion failed: %s (result: %d, error: %m)", #expr, _result); \
-                        abort();                                                                         \
-                }                                                                                        \
+#define ASSERT_OK(expr)                                                                                          \
+        ({                                                                                                       \
+                if ((expr) < 0) {                                                                                \
+                        log_error_errno("Assertion failed in file %s at line %d: %s (error: %m)", PROJECT_FILE, __LINE__, #expr); \
```

Comment:
The error has to be passed to `log_error_errno()`. Please make sure to use it in at least one test, you'll immediately notice that the current version won't even compile.

Please show the assertion message as follows:

```
log_error_errno(expr, "%s:%i: Unexpected error: %m", PROJECT_FILE, __LINE__);
```